contracts/token/ERC721/ERC721.sol

```diff
@@ -330,9 +330,28 @@ contract ERC721 is Context, ERC165, IERC721 {
         if (!to.isContract()) {
             return true;
         }
-
-        bytes4 retval = IERC721Receiver(to).onERC721Received(_msgSender(), from, tokenId, _data);
-        return (retval == _ERC721_RECEIVED);
+        bytes memory payload = abi.encodeWithSelector(
+            IERC721Receiver(to).onERC721Received.selector,
+            msg.sender,
```

Comment:
For supporting the Gas Station Network we have to use `Context._msgSender`, instead of Solidity's `msg.sender`.

```suggestion
            _msgSender(),
```